server/aesdsocket.c

```diff
@@ -197,48 +236,62 @@ void* thread_function(void* thread_param){
 				newline = strchr(buf, '\n');	//check for newline character
 		}while(newline == NULL); 
 
+
 		pthread_mutex_lock(&mutex);										//lock the write functionality
 		//write into the file
         writeret = write(storefd, buf2, buf2_size);
-            if(writeret == -1){
+            if(writeret != buf2_size){
                 syslog(LOG_ERR, "Write error!!");
+				pthread_mutex_unlock(&mutex);
 				return NULL;
-            }
-		pthread_mutex_unlock(&mutex);	
-
+            }	
 
+#if USE_AESD_CHAR_DEVICE
+		end_of_file += writeret;
```

Comment:
@Mich2899 I'd suggest not using different logic here for using/not using aesdchar device.  I think you could simplify the logic, remove alternate implementations/extra varaibles, and remove assuptions about total size of all packets sent if you just read and send line by line (packet by packet)